bundles/Expense/Controller/ExpenseController.php

```
@@ -61,6 +62,35 @@ public function new(Request $request, TranslatorInterface $translator, JiraServi
     {
         $projects = $jiraService->getProjects();
 
+        // Get project and issue from query string.
+        $selectedProject = null;
+        $selectedIssue = null;
+
+        if (null !== $projectId = $request->get('project')) {
+            // Find project by id or key.
+            $selected = array_filter(
+                $projects,
+                function ($project) use ($projectId) {
+                    return $projectId === $project->id || $projectId === $project->key;
+                }
+            );
+            $selectedProject = reset($selected);
+
+            if (null !== $selectedProject) {
+                if (null !== $issueId = $request->get('issue')) {
+                    try {
+                        $selectedIssue = $jiraService->getIssue($issueId);
+                        // Check that the selected issue belongs to the selected project.
+                        if (!isset($selectedIssue->fields->project->id, $selectedProject->id)
+                            || $selectedIssue->fields->project->id !== $selectedProject->id) {
+                            $selectedIssue = null;
+                        }
+                    } catch (ClientException $exception) {
```

Please leave a comment if ignoring the exception is on purpose.